client/dashboard/components/lazy-modal/index.tsx

```diff
@@ -0,0 +1,25 @@
+import { lazy, Suspense, type ComponentType } from 'react';
+
+// If you're getting a type error here, maybe check that the module you are dynamically
+// importing (with `import()`) has a default export that is a React component.
+type ExtractComponentProps< T > = T extends () => Promise< { default: infer U } >
+	? U extends ComponentType< infer P >
+		? P
+		: never
+	: never;
+
+/**
+ * Wraps React's built-in `lazy` function. No fallback is shown during loading, which
+ * should be suitable when lazy-loading modals specifically.
+ */
+export function lazyModal< F extends Parameters< typeof lazy >[ 0 ] >( importFn: F ) {
```

Comment:
I think this is a general-purpose function for creating a lazy-loaded component, and it doesn't have any internal logic specific to modals. It would be better to name it something more generic, like createLazyComponent.

Reply:
The part that makes it specifically for modals is that it has made the design decision about what the loading behaviour should be. Other async components it would be appropriate to fallback to a much higher level `<Suspense>` wrapper (although we're not used to doing that sort of thing in Calypso). Other lazy components it would be much better to show a spinner.

But for modals we know we can say that it is better to just delay showing the modal for a few milliseconds.

But I think you're right about not having a wrapper. At least until we have a very well established design pattern. It makes sense to not have a wrapper, and to have each UI explicitly say what loading behaviour is right for them.